mantidproject/mantid
# Workbench: Notification service and Notifications on errors

**Description of work.**

Added a NotificationService QtWidget, and set the messagedisplay in workbench to show a notification when it receives an error or higher.  This was asked for as just errors in the log can be hard to spot. 
Also added the ability to turn off notifications using the workbench properties or the properties file.

**Report to:** James Lord

**To test:**
**Important:** The implementation of notifications varies across different OS's, so this needs testing on:

1. Start workbench (there are no notifications set in MPlot)
1. open File->Settings, check that notifications are on
![image](https://user-images.githubusercontent.com/1017173/70536158-216ab480-1b56-11ea-81c6-9c6078c45c34.png)
1. run some things that will produce an error.  Incorrect python is the easiest to do.
```python
# The following line helps with future compatibility with Python 3
# print must now be used as a function, e.g print('Hello','World')
from __future__ import (absolute_import, division, print_function, unicode_literals)

# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *

import matplotlib.pyplot as plt

import numpy a
```
1. Repeat with notifications turned off
1. Check that the setting is preserved across workbench sessions.


Fixes #27305

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
